### PR TITLE
Merge master

### PR DESCRIPTION
**Pull request by pacu**
I had to make TestCoordinator.build async and I found a workaround. Please comment on that!
image

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the app and try the changes? 
- [x] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [x] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: Tests/TestUtils/TestCoordinator.swift [hunk 1]
```diff
@@ -219,19 +225,25 @@ extension TestCoordinator {
...
-            dataDb: config.dataDb,
-            downloadBatchSize: config.downloadBatchSize,
-            retries: config.retries,
-            maxBackoffInterval: config.maxBackoffInterval,
-            rewindDistance: config.rewindDistance,
-            walletBirthday: config.walletBirthday,
-            saplingActivation: config.saplingActivation,
-            network: config.network
-        )
+        Task {
```
Comment (Chlup) [inline Tests/TestUtils/TestCoordinator.swift:228]: I have same comment about this as in `BalanceTests.swift`.

File: Tests/TestUtils/XCAsyncTestCase.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        }
+    }
+
+    func wait(asyncBlock: @escaping (() async throws -> Void)) {
+        let semaphore = DispatchSemaphore(value: 0)
+        Task.init {
+            try await asyncBlock()
+            semaphore.signal()
+        }
+        semaphore.wait()
```
Comment (Chlup) [inline Tests/TestUtils/XCAsyncTestCase.swift:40]: Nicely done!

File: ZcashLightClientKit.podspec [hunk 3]
```diff
@@ -1,6 +1,10 @@
 Pod::Spec.new do |s|
     s.name             = 'ZcashLightClientKit'
-    s.version          = '0.17.0-alpha.1'
+<<<<<<< HEAD
```
Comment (Chlup) [inline ZcashLightClientKit.podspec:3]: Forgotten git merge marks.

**Review by Chlup — CHANGES_REQUESTED** [review]

File: Tests/DarksideTests/BalanceTests.swift [hunk 4]
```diff
@@ -30,12 +30,14 @@ class BalanceTests: XCTestCase {
...
-        )
+        Task{ @MainActor [self] in
+            self.coordinator = try await TestCoordinator(
+                seed: seedPhrase,
+                walletBirthday: birthday,
+                channelProvider: ChannelProvider(),
+                network: network
+            )
+        }
         try coordinator.reset(saplingActivation: 663150, branchID: "e9ff75a6", chainName: "main")
```
[the remark this answers is not in the document]
Comment (pacu) [reply Tests/DarksideTests/BalanceTests.swift:41]: moved to asyncSetUp

File: Sources/ZcashLightClientKit/Block/Processor/CompactBlockProcessor.swift [hunk 5]
```diff
@@ -312,51 +312,12 @@ public class CompactBlockProcessor {
...
-        }
-
-        public func get() -> Bool {
-            lock.lock()
-            defer { lock.unlock() }
-            return needsToStartScanning
-        }
-    }
-
-    private let needsToStartScanningWhenStopped = ThreadSafeNeedsToStartScanning()
```
[the remark this answers is not in the document]
Comment (pacu) [reply Sources/ZcashLightClientKit/Block/Processor/CompactBlockProcessor.swift:358]: I thought we could remove it when this became an actor. can you clarify?

[the remark this answers is not in the document]
Comment (pacu) [reply Sources/ZcashLightClientKit/Block/Processor/CompactBlockProcessor.swift:358]: I think I did what you meant. Please check 🙏

[the remark this answers is not in the document]
Comment (Chlup) [reply Sources/ZcashLightClientKit/Block/Processor/CompactBlockProcessor.swift:358]: I am sorry. I described it wrong. I meant that `ThreadSafeNeedsToStartScanning` data type can be gone. You did exactly what I meant. 👍

**Review by Chlup — APPROVED** [review]

**Review by LukasKorba — APPROVED** [review]
LGTM, I haven't caught anything else than Michal

**Pull request merged**